plutus-tx/src/PlutusTx/Prelude.hs

```diff
@@ -178,8 +181,8 @@ import Prelude qualified as Haskell (return, (=<<), (>>), (>>=))
 
 {-# INLINABLE check #-}
 -- | Checks a 'Bool' and aborts if it is false.
-check :: Bool -> ()
-check b = if b then () else traceError checkHasFailedError
+check :: Bool -> BI.BuiltinUnit
+check b = if b then BI.unitval else traceError checkHasFailedError
```

Comment:
This is big change. This means that we can never "backport" anymore the NonUnitReturnValue error to also PV1,PV2 (if we ever decide that we want that)

Reply:
Not exactly, but expecting `BuiltinData` instead of `BuiltinUnit` as the final result value would be more uniform, because a function would be of type `BuiltinData -> BuiltinData` (for v3) . And it would still leave open the possibility in the future to store/cache on-chain  the result of the computations serialised (a feature that nobody wants I guess)

Reply:
> This is big change. This means that we can never "backport" anymore the InvlalidReturnValue error to also PV1,PV2 (if we ever decide that we want that)

I'm not sure I understand - this is just a Plutus Tx helper function, how is this a big change?

But in any case, I don't think we can ever backport this to V1 and V2 anyway, since there are already lots of scripts onchain that don't return BuiltinUnit, and those script must continue to work.

Reply:
> the serialised result (BuiltinData) of the computations (a feature that nobody wants I guess).

We can have more discussion on this for Plutus V4. The motivation is not super clear to me..

Reply:
Not very important. I just assumed that  Tx.check was all these years using `BuiltinUnit` as the result already. So I had hopes that we could  backport this BuiltinUnit-result-restriction also to PV1,PV2 somewhere in the future. But it is clear to me now that it definitely cannot be backported.